pkg/querier/queryrange/split_by_interval.go

```diff
@@ -193,29 +203,21 @@ func (h *splitByInterval) Do(ctx context.Context, r queryrange.Request) (queryra
 		})
 	}
 
-	resps, err := h.Process(ctx, h.limits.MaxQueryParallelism(userid), int64(lokiRequest.Limit), input)
+	resps, err := h.Process(ctx, h.limits.MaxQueryParallelism(userid), limit, input)
 	if err != nil {
 		return nil, err
 	}
 	return h.merger.MergeResponse(resps...)
 }
 
-func splitByTime(r *LokiRequest, interval time.Duration) []queryrange.Request {
+func splitByTime(req queryrange.Request, interval time.Duration) []queryrange.Request {
 	var reqs []queryrange.Request
-	for start := r.StartTs; start.Before(r.EndTs); start = start.Add(interval) {
-		end := start.Add(interval)
-		if end.After(r.EndTs) {
-			end = r.EndTs
+	for start := req.GetStart(); start < req.GetEnd(); start = start + interval.Milliseconds() {
```

Comment:
This change limits us to use only `millisecond`. Since, `GetStart` and `GetEnd` methods converts the time to `milliseconds` we won't able to measure anything below `millisecond`

Reply:
Great point, I didn't think of that.

I think your original approach is better :)